File: LeetCode/Algo/DP/fib.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0:
            return 0
        res = [0] * (n+1)
        res[0] = 0
        res[1] = 1
        for i in range(2, n+1):
            res[i] = res[i-1] + res[i-2]
        return res[n]
    
solve = Solution()
print(solve.fib(0))

# 0 1 1 2 3 5 

File: LeetCode/Algo/DP/fib.java
import java.util.Scanner;
class Solution {
    public long fib(int n) {
       long pre = 1; //  f1 = 1
       long curr = 1; // f2 = 1
       for(int i = 3; i <= n; i++){
           long result = pre + curr;
           pre = curr;
           curr = result;
       }
       return curr;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Solution solve = new Solution();
        System.out.println(solve.fib(n));
    }
}
// 1 1 2 3 5